client/client_test.go

```diff
@@ -5879,6 +5880,225 @@ func testCallInfo(t *testing.T, sb integration.Sandbox) {
 	require.NoError(t, err)
 }
 
+func testExportAnnotations(t *testing.T, sb integration.Sandbox) {
+	requiresLinux(t)
+	c, err := New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	amd64 := platforms.MustParse("linux/amd64")
+	arm64 := platforms.MustParse("linux/arm64")
+	ps := []ocispecs.Platform{amd64, arm64}
+
+	frontend := func(ctx context.Context, c gateway.Client) (*gateway.Result, error) {
+		res := gateway.NewResult()
+		expPlatforms := &exptypes.Platforms{
+			Platforms: make([]exptypes.Platform, len(ps)),
+		}
+		for i, p := range ps {
+			st := llb.Scratch().File(
+				llb.Mkfile("platform", 0600, []byte(platforms.Format(p))),
+			)
+
+			def, err := st.Marshal(ctx)
+			if err != nil {
+				return nil, err
+			}
+
+			r, err := c.Solve(ctx, gateway.SolveRequest{
+				Definition: def.ToPB(),
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			ref, err := r.SingleRef()
+			if err != nil {
+				return nil, err
+			}
+
+			_, err = ref.ToState()
+			if err != nil {
+				return nil, err
+			}
+
+			k := platforms.Format(p)
+			res.AddRef(k, ref)
+
+			expPlatforms.Platforms[i] = exptypes.Platform{
+				ID:       k,
+				Platform: p,
+			}
+		}
+		dt, err := json.Marshal(expPlatforms)
+		if err != nil {
+			return nil, err
+		}
+		res.AddMeta(exptypes.ExporterPlatformsKey, dt)
+
+		res.AddMeta(exptypes.ExporterAnnotationIndexKey("gi"), []byte("generic index"))
+		res.AddMeta(exptypes.ExporterAnnotationIndexDescriptorKey("gid"), []byte("generic index descriptor"))
+		res.AddMeta(exptypes.ExporterAnnotationManifestKey(nil, "gm"), []byte("generic manifest"))
+		res.AddMeta(exptypes.ExporterAnnotationManifestDescriptorKey(nil, "gmd"), []byte("generic manifest descriptor"))
+		res.AddMeta(exptypes.ExporterAnnotationManifestKey(&amd64, "m"), []byte("amd64 manifest"))
+		res.AddMeta(exptypes.ExporterAnnotationManifestKey(&arm64, "m"), []byte("arm64 manifest"))
+		res.AddMeta(exptypes.ExporterAnnotationManifestDescriptorKey(&amd64, "md"), []byte("amd64 manifest descriptor"))
+		res.AddMeta(exptypes.ExporterAnnotationManifestDescriptorKey(&arm64, "md"), []byte("arm64 manifest descriptor"))
+
+		return res, nil
+	}
+
+	// testing for image exporter
+
+	registry, err := sb.NewRegistry()
```

Comment:
Is the registry used by this test at all? Looks like we are reading from containerd imagestore.